PR for feature branch

praz99 opened the pull request:
In this project we

- added user sessions.

- generated controller for user, event, and event attendance.

- created associations between users and events.

- added RSpec test for model validations and associations.

- added RSpec and Capybara for testing features and integrations.

praz99 committed:
add User model

praz99 committed:
add User controller and corresponding routes
routes for #new, #create and #show actions

praz99 committed:
add views for user methods

praz99 committed:
add authentication for user
add login with only username
add signup page
generate session controller
use session for authentication
add views for various methods

okikiola11 committed:
Generate the event controller and migration file

okikiola11 committed:
Add the events routes
Generate the event model
Create the add_creator_to_event file
Generate the associations between events and user
Modify the user controller to show all events

okikiola11 committed:
Generate the addfieldstoevents file

okikiola11 committed:
Add the show method to the event_controller
The show method displays the creator_id of a user event

praz99 committed:
Update the show event page
display event parameters
display all events in event index page

praz99 committed:
create relationship between event and event_attendings

praz99 committed:
change index page for events
make index page show all the events
make user show page display all the events created by user

praz99 committed:
add new method on event_controller

okikiola11 committed:
Add Add the created events by a user to the User controller show method
Updated the routes to accept the event routes

okikiola11 committed:
Add the model for the previously attended events by the user
Add the upcoming event to be attended by the user
Add the model to accept to previously attended events and upcoming events to accept the events_date

okikiola11 committed:
Modify the event show page

okikiola11 committed:
Add the upcoming and previously created events to accept date
Update the UI of the view pages

okikiola11 committed:
Pass the event parameter to take route path in user/show.html

okikiola11 committed:
Edit the route

praz99 committed:
add back button for create index page

praz99 committed:
setup rspec and capybara gems

praz99 committed:
add test for user and event model validations

praz99 committed:
add tests for model associations

okikiola11 committed:
Add new gems for testing
Add the support file for factory settings
Add the controller spec file

okikiola11 committed:
Add the gems for testing
Add the feature test and configurations

okikiola11 committed:
Add the feature test for authentication

okikiola11 committed:
Add the feature test for event_management.rb

okikiola11 committed:
Add the feature test for events
Create the event_management_spec to mange events

okikiola11 committed:
Fix the authentication test

okikiola11 committed:
Update the README.md

praz99 committed:
add linters and fix lint errors

lekegitrepo commented on db/schema.rb:
```diff
@@ -0,0 +1,42 @@
+    t.datetime "events_date"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.integer "creator_id"
+    t.string "title"
+    t.text "description"
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string "name"
+    t.string "username"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
+
+  add_foreign_key "event_attendings", "events", column: "attended_events_id"
+  add_foreign_key "event_attendings", "users", column: "event_attendee_id"
```
- The following columns of the database should be index
  - creator_id
  - event_attendings

lekegitrepo commented on app/views/shared/_error_messages.html.erb:
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
+      <li class="text-danger"><%= msg %></li>
+    <% end %>
+    </ul>
+  </div>
+<% end %>
```
- View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
- There should be absolutely NO logic in views.

lekegitrepo commented on app/views/layouts/_header.html.erb:
```diff
@@ -0,0 +1,18 @@
+    <div class="navbar-header">
+      <%= link_to "Events", root_path, class: 'mx-2 navbar-brand' %>
+    </div>
+    <% if logged_in? %>
+        <nav class="navbar navbar-light bg-light justify-content-between">
+          <%= link_to current_user.name, user_path(current_user) %>
+          <%= link_to "log out", logout_path, class: 'mx-2', method: :delete %>
+        </nav>
+      <% else %>
+        <nav class="navbar navbar-light bg-light justify-content-between">
+          <%= link_to "Log in", login_path, class: 'mx-2' %>
+          <%= link_to "Sign up", signup_path, class: 'mx-2' %>
+        </nav>
+      <% end %>
```
- View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
- There should be absolutely NO logic in views.

lekegitrepo commented on app/views/event_attendings/new.html.erb:
```diff
@@ -0,0 +1,12 @@
+<h1>Attend Event</h1>
+<%= form_for [@event, @event.event_attendings.build] do |f| %>
+<!--<%= f.collection_check_boxes :event_attendee_id, User.all, :id, :username %>-->
```
- It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

lekegitrepo reviewed (changes requested):
### Status: Request changes :recycle:
#### Hi @praz99 and @okikiola11 :clap: :clap:
#### You have done a good job so far, however, there are still some requirements you need to fix before this project can be approved. Check my comments below.

**_- The event's show should page displays the list of attendants ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/login_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

**_- The user should provide email ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_can_create_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))_**

**_Check the remaining comments below :point_down:_** 

#### _Reviewed by Temitayo Adeleke._

lekegitrepo commented on app/controllers/event_attendings_controller.rb:
```diff
@@ -0,0 +1,30 @@
+# class EventAttendingsController < ApplicationController
+#   def new
+#     @event = Event.find(params[:event_id])
+#     @event_attending = EventAttending.new
+#   end
+
+#   def create
+#     @event = Event.find(params[:event_id])
+#     @event_attending = EventAttending.new
+#     if params[:event_attending]
+#       params[:event_attending][:event_attendee_id].each do |x|
+#         @event_attending.event_attendee = User.find(x.to_i)
+#         @event_attending.attended_events = @event
+#       end
+#       flash[:success] = 'Event successfully created'
+#       redirect_to user_path(@event.creator)
+#     else
+#       flash[:error] = 'Something went wrong'
+#       render 'new'
+#     end
+#   end
+
+#   def show
+#     @event_attending = EventAttending.find(params[:id])
+#   end
+
+#   def att_params
+#     params.raquire(:event_attending).permit(:event_attendee_id, :attended_events_id)
+#   end
+# end
```
- It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

praz99 committed:
remove comments

praz99 committed:
add link to show single event from event index

praz99 committed:
generate new migration
add index for events table in creator_id column

praz99 committed:
Remove logic from views
create different partials for each logic condition,
render the partials from methods within view helpers.

praz99 committed:
fix lint error

rOluochKe reviewed (approved):
Hello Team
### Well done, you got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project, good luck with the next one.
- Happy coding!
